### PR TITLE
Small fixes to ensure dateClimbed as date works

### PR DESCRIPTION
**Pull request by zichongkao**
Tick can be added (This was before validation. Today is 6/8. Adding 7/8 shouldn't have been possible):
image

Date validation works:
image

### CODE REVIEW

File: src/components/users/TickCard.tsx [hunk 1]
```diff
@@ -8,7 +8,7 @@ interface Props{
   ticks: TickType[]
   setTicks: Function
   tickId: string
-  dateClimbed: Date
+  dateClimbed: number
```
Comment (zichongkao) [inline src/components/users/TickCard.tsx:11]: The backend returns the date as a Unix timestamp.

File: src/components/users/TickForm.tsx [hunk 2]
```diff
@@ -91,7 +92,7 @@ export default function TickForm ({ open, setOpen, setTicks, ticks, isTicked, cl
       userId: session.data?.user.metadata.uuid,
       style: style.name,
       attemptType: attemptType.name,
-      dateClimbed: dateClimbed,
+      dateClimbed: new Date(Date.parse(`${dateClimbed}T00:00:00`)), // Date.parse without timezone converts dateClimbed into local timezone.
```
Comment (zichongkao) [inline src/components/users/TickForm.tsx:95]: The backend accepts anything that `new Date(...)` will accept, so we could pass in a number such as `Date.parse(..)` alone. However, the `Yup.Date()` validation is a bit easier to read if these are dates.

File: src/components/users/TickForm.tsx [hunk 3]
```diff
@@ -118,9 +119,16 @@ export default function TickForm ({ open, setOpen, setTicks, ticks, isTicked, cl
         }
       })
       .catch((error) => {
```
Comment (zichongkao) [inline src/components/users/TickForm.tsx:121]: Yup validations throw errors too, and these don't have `grapjhQLErrors` fields.

File: src/pages/api/user/ticks.ts [hunk 4]
```diff
@@ -60,12 +79,12 @@ const handler: NextApiHandler<any> = async (req, res) => {
         ret.forEach((tick) => {
           const newTick: Tick = {
             name: tick.Route,
-            notes: tick.notes,
```
Comment (zichongkao) [inline src/pages/api/user/ticks.ts:63]: This is a bug -- the CSV I downloaded uses "Notes":
image

**Comment by zichongkao** [comment]
I didn't test MP import locally. But given that we know the CSV inputs i.e.  dates in the format "2022-01-01", and the fact that `new Date('2022-01-01T00:00:00')` results in the correct input to our backend, I'm confident that that change won't cause problems.

File: src/components/users/TickForm.tsx [hunk 5]
```diff
@@ -24,7 +24,8 @@ const TickSchema = Yup.object().shape({
   attemptType: Yup.string()
     .required('Please choose an ascent type'),
   dateClimbed: Yup.date()
-    .required('Please include a date'),
+    .required('Please include a date')
+    .max(new Date(), 'Please include a date in the past'),
```
Comment (musoke) [inline src/components/users/TickForm.tsx:28]: Is the date check done in the user's local timezone?  Looks like it from the comment on L#95?

**Review by musoke — APPROVED** [review]

**Review by vnugent — APPROVED** [review]

**Pull request merged**